InMobi/src/main/java/com/mopub/mobileads/InMobiInterstitial.kt

```diff
@@ -0,0 +1,171 @@
+                }
+
+                override fun onAdWillDisplay(inMobiInterstitial: InMobiInterstitial) {
+                    MoPubLog.log(adNetworkId, AdapterLogEvent.WILL_APPEAR, ADAPTER_NAME)
+                }
+
+                override fun onAdDisplayed(ad: InMobiInterstitial, info: AdMetaInfo) {
+                    MoPubLog.log(adNetworkId, AdapterLogEvent.CUSTOM, ADAPTER_NAME,
+                            "InMobi interstitial ad displayed")
+                    MoPubLog.log(adNetworkId, AdapterLogEvent.SHOW_SUCCESS, ADAPTER_NAME)
+                    mInteractionListener?.onAdShown()
+                    mInteractionListener?.onAdImpression()
+                }
+
+                override fun onAdClicked(inMobiInterstitial: InMobiInterstitial,
+                                         params: Map<Any?, Any?>?) {
+                    MoPubLog.log(adNetworkId, AdapterLogEvent.CLICKED, ADAPTER_NAME)
+                    mInteractionListener?.onAdClicked()
+                }
+
+                override fun onAdDisplayFailed(inMobiInterstitial: InMobiInterstitial) {
+                    onInMobiAdFailWithEvent(AdapterLogEvent.SHOW_FAILED, adNetworkId,
+                            MoPubErrorCode.FULLSCREEN_SHOW_ERROR,
+                            "InMobi interstitial show failed",
+                            ADAPTER_NAME, null, mInteractionListener)
+                }
+
+                override fun onAdDismissed(inMobiInterstitial: InMobiInterstitial) {
+                    MoPubLog.log(adNetworkId, AdapterLogEvent.CUSTOM, ADAPTER_NAME,
+                            "InMobi interstitial ad dismissed")
+                    MoPubLog.log(adNetworkId, AdapterLogEvent.DID_DISAPPEAR, ADAPTER_NAME)
+                    mInteractionListener?.onAdDismissed()
+                }
+
+                override fun onRewardsUnlocked(inMobiInterstitial: InMobiInterstitial,
+                                               rewards: Map<Any, Any>?) {
+                    // no-op
+                }
+            })
+
+            MoPubLog.log(adNetworkId, AdapterLogEvent.LOAD_ATTEMPTED, ADAPTER_NAME)
+            val adMarkup = extras[DataKeys.ADM_KEY]
+            if (adMarkup != null) {
+                MoPubLog.log(AdapterLogEvent.CUSTOM, ADAPTER_NAME,
+                        "Ad markup for InMobi interstitial ad request is present. Will make Advanced Bidding ad request " +
+                                "using markup: " + adMarkup)
+                mInMobiInterstitial?.run {
+                    load(adMarkup.toByteArray())
+                }
+            } else {
+                MoPubLog.log(AdapterLogEvent.CUSTOM, ADAPTER_NAME,
+                        "Ad markup for InMobi interstitial ad request is not present. Will make traditional ad request ")
+                mInMobiInterstitial?.run {
+                    setExtras(InMobiAdapterConfiguration.inMobiTPExtras)
+                    load()
+                }
```

Comment:
Its good to keep null checks and it looks like it is not going to be `null` but if by chance it is null then it won't end up giving any callbacks. Make sure you either handle those or use `!!`.